provider/googleProvider.js

```diff
@@ -92,6 +92,29 @@ class GoogleProvider {
           },
           additionalProperties: false,
         },
+        cloudFunctionSecretEnvironmentVariables: {
+          type: 'object',
+          properties: {
+            key: {
+              type: 'string',
+              minLength: 1,
+              pattern: '^[a-zA-Z0-9_]+$',
+            },
+            projectId: {
+              type: 'string',
+              minLength: 1,
+            },
+            secret: {
+              type: 'string',
+              pattern: '^[a-zA-Z0-9_-]+$',
+            },
+            version: {
+              type: 'string',
+              pattern: '^(latest|[0-9.]+)$',
+            },
+          },
+          required: ['key', 'secret', 'version'],
+        },
```

Comment:
I misread the original validation pattern.

This will allow for the key to be defined in a similar way to the standard ENV vars:

```suggestion
        cloudFunctionSecretEnvironmentVariables: {
          type: 'object',
          patternProperties: {
            '^[a-zA-Z0-9_]+$': {
              type: 'object',
              properties: {
                projectId: {
                  type: 'string',
                  minLength: 1,
                },
                secret: {
                  type: 'string',
                  pattern: '^[a-zA-Z0-9_-]+$',
                },
                version: {
                  type: 'string',
                  pattern: '^(latest|[0-9.]+)$',
                },
              },
              required: ['secret', 'version'],
            },
          },
        },
```

The resulting config would look something like:

```yaml
…
secrets:
  SOME_SECRET:
    secret: abcd1234
    version: latest
  SOME_OTHER_PROJECTS_SECRET:
    projectId: another-project
    secret: super$secret$string
    version: latest
…
```

But that raises the question whether it is sensible to use the config to set secrets, or if it should just be used to retrieve secrets. If it was just for retrieval, then the config would not include the secret itself.

GCF has a way to enable access to secrets as ENV vars - which is probably the neatest path to bringing secrets into a function - https://cloud.google.com/functions/docs/configuring/secrets 

Perhaps we should be exploring a way to enable the access to particular secrets as ENV vars via the config, something like:

```
…
environment:
  NOT_A_SECRET_VAR: foo-bar
  SECRET_VAR: $secret:project/PROJECT_ID/secrets/SECRET_VAR_NAME@SECRET_VERSION
…
```

Reply:
`cloudFunctionSecretEnvironmentVariables` in the Deployment Manager config does exactly that, sets them as environment variables in the runtime environment of the function.

I don't think it's a good idea to invent a syntax over the 3 or 4 options per secret.